📍 Add connecting to local chain example (#630)
TrueFiEng/useDApp: packages/docs/docs/guides/local-chain.mdx

(3) 0xc2c5eefc5b84d4b0a12d6384f502e31e771c1e1980f253dccc4ee51e45ffe08c
(4) 0x2a932d7ec4abe76914cee7e95b287946a8fc9f1035d367201f42f8d2f12126d3
(5) 0x75c6cb6c8e2b822547e19990d3329d81f85014dfb5a22374ab41b777ebe03ba4
(6) 0x33cd5cd08cefd3027d79ad72d060b5162dc4284734dabc747644e06f90a33829
(7) 0x45279b730ca8b4fcf6db360da6136ca5fe33e88e621b3a2012123add4093cd39
(8) 0x01fea72899b1d02befcde988e9d1f152ce8ffb4872cdfdeb2676ad3fcf1746f2
(9) 0x40e7e7e7c53b885f86cff7479ba42dc48c661a98e802867462e4f4e0283cb625

HD Wallet
==================
Mnemonic:      ribbon twice crisp coconut spot tube cushion upset sunny spell riot empower
Base HD Path:  m/44'/60'/0'/0/{account_index}

Default Gas Price
==================
2000000000

BlockGas Limit
==================
30000000

Call Gas Limit
==================
50000000

Chain Id
==================
1337

RPC Listening on 127.0.0.1:8545
```

## Configuration

Add ganache address to configuration.

```tsx
  import { Localhost } from '@usedapp/core'

  const config = {
    readOnlyChainId: Localhost.chainID,
    readOnlyUrls: {
      [Localhost.chainID]: 'http://127.0.0.1:8545',
    },
  }
```

## Reading from blockchain

> To read from blockchain just add a `queryOptions` object to specified call.

Reviewer: ```suggestion
To read from blockchain, just add a `queryOptions` object to specified call.
```

> You do not have to deploy multicall, because useDapp do it itself.

Reviewer: ```suggestion
You do not have to deploy multicall, because useDapp does it automatically.
```

---
sidebar_position: 6

---

# Connecting to local Chain

> In this tutorial, we will go through the steps of connecting to a local Chain.

Reviewer: ```suggestion
In this tutorial, we will go through the steps of connecting to a local Chain.
Connecting to a local Chain can be useful for local development of DApps for a number of reasons:

- Transactions are faster,
- Possibility to work offline,
- Not relying on testnet faucets.
```

> In this tutorial, we will go through the steps of connecting to a local Chain.

Reviewer: ```suggestion
In this tutorial, we will go through the steps of connecting to a local Chain using Ganache <AND LINK>.
```

## Prerequisites

The tutorial assumes the user has already started with the basics of `useDApp`.

See the [Getting Started](../getting-started.mdx) guide if you are a new user.

## Start ganache

> Ganache is a personal blockchain for rapid Ethereum and Corda distributed application development. 

Reviewer: ```suggestion
Ganache is a personal blockchain for rapid Ethereum distributed application development. 
```

You can use Ganache across the entire development cycle; enabling you to develop, deploy, and test your dApps in a safe and deterministic environment. 

> Ganache comes in two flavors: a UI and CLI.

Reviewer: ```suggestion
```

(3) 0xc2c5eefc5b84d4b0a12d6384f502e31e771c1e1980f253dccc4ee51e45ffe08c
(4) 0x2a932d7ec4abe76914cee7e95b287946a8fc9f1035d367201f42f8d2f12126d3
(5) 0x75c6cb6c8e2b822547e19990d3329d81f85014dfb5a22374ab41b777ebe03ba4
(6) 0x33cd5cd08cefd3027d79ad72d060b5162dc4284734dabc747644e06f90a33829
(7) 0x45279b730ca8b4fcf6db360da6136ca5fe33e88e621b3a2012123add4093cd39
(8) 0x01fea72899b1d02befcde988e9d1f152ce8ffb4872cdfdeb2676ad3fcf1746f2
(9) 0x40e7e7e7c53b885f86cff7479ba42dc48c661a98e802867462e4f4e0283cb625

HD Wallet
==================
Mnemonic:      ribbon twice crisp coconut spot tube cushion upset sunny spell riot empower
Base HD Path:  m/44'/60'/0'/0/{account_index}

Default Gas Price
==================
2000000000

BlockGas Limit
==================
30000000

Call Gas Limit
==================
50000000

Chain Id
==================
1337

RPC Listening on 127.0.0.1:8545
```

## Configuration

Add ganache address to configuration.

```tsx
  import { Localhost } from '@usedapp/core'

  const config = {
    readOnlyChainId: Localhost.chainID,
    readOnlyUrls: {
      [Localhost.chainID]: 'http://127.0.0.1:8545',//paste ganache address
    },
  }
```

> ## Deploy multicall

Reviewer: This should not be needed.
If `isLocalChain` is true, usedapp is supposed to deploy a multicall automatically.

Check that it works.

Deploy multicall contract to the ganache and paste its address to file local.ts

```tsx
export const Localhost: Chain = {
  chainId: 1337,
  chainName: 'Localhost',
  isTestChain: true,
  isLocalChain: true,
  multicallAddress: '0xmulticallAddress',//paste local multicall address
  getExplorerAddressLink: () => '',
  getExplorerTransactionLink: () => '',
}
```

## Reading from blockchain

> To read from blockchain just add the queryOptions object to specified call.

Reviewer: ```suggestion
To read from blockchain just add a `queryOptions` object to specified call.
```

### Example

```tsx
const etherBalance = useEtherBalance(account, { chainId: Localhost.chainId })
const tokenBalance = useTokenBalance(tokenAddress, account, { chainId: Localhost.chainId })
```

## Summary

> In this tutorial, we went through the steps required for using useDApp with a local chain.

Reviewer: ```suggestion
In this tutorial, we went through the steps required for using useDApp with a local chain using Ganache.
```

> We modify the local chain by adding local multicall address.

Reviewer: ```suggestion
```
